OsmAnd/src/net/osmand/plus/settings/fragments/AuthorizeFragment.java

```diff
@@ -0,0 +1,451 @@
+package net.osmand.plus.settings.fragments;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.Spannable;
+import android.text.SpannableString;
+import android.text.TextWatcher;
+import android.text.style.ForegroundColorSpan;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.view.ViewGroup;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.StringRes;
+import androidx.appcompat.widget.Toolbar;
+import androidx.core.content.ContextCompat;
+import androidx.core.view.ViewCompat;
+import androidx.fragment.app.FragmentActivity;
+import androidx.fragment.app.FragmentManager;
+
+import com.google.android.material.appbar.AppBarLayout;
+
+import net.osmand.AndroidUtils;
+import net.osmand.PlatformUtil;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.UiUtilities;
+import net.osmand.plus.UiUtilities.DialogButtonType;
+import net.osmand.plus.backup.BackupHelper;
+import net.osmand.plus.backup.BackupHelper.OnRegisterDeviceListener;
+import net.osmand.plus.backup.BackupHelper.OnRegisterUserListener;
+import net.osmand.plus.base.BaseOsmAndFragment;
+import net.osmand.plus.development.TestBackupActivity;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.settings.backend.OsmandSettings;
+import net.osmand.plus.settings.fragments.BaseSettingsFragment.SettingsScreenType;
+import net.osmand.plus.wikipedia.WikipediaDialogFragment;
+import net.osmand.util.Algorithms;
+
+import org.apache.commons.logging.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class AuthorizeFragment extends BaseOsmAndFragment {
+
+	private static final Log log = PlatformUtil.getLog(AuthorizeFragment.class);
+
+	public static final String TAG = AuthorizeFragment.class.getSimpleName();
+
+	private static final String OSMAND_EMAIL = "support@osmand.net";
+	private static final String OSMAND_DOCS = "https://docs.osmand.net/en/main@latest/osmand/purchases/android";
+
+	private static final String LOGIN_DIALOG_TYPE_KEY = "login_dialog_type_key";
+
+	private static final int VERIFICATION_CODE_EXPIRATION_TIME_MIN = 60 * 1000; // 10 minutes
+
+	private static final int ERROR_CODE_USER_IS_NOT_REGISTERED = 103;
```

Comment:
Move to BackupHelper. Copy all constants from [server](https://github.com/osmandapp/OsmAnd-tools/blob/152c78fd6b6b863fbf4f618ae38b68bb8565acf6/java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/UserdataController.java#L63) without
 ERROR_CODE_PREMIUM_USERS. Call SERVER_ERROR_CODE_...